processing/src/main/java/org/apache/druid/java/util/common/concurrent/Execs.java

```diff
@@ -159,6 +162,10 @@ public static ExecutorService newBlockingThreaded(
           @Override
           public void rejectedExecution(Runnable r, ThreadPoolExecutor executor)
           {
+            if (executor.isShutdown()) {
+              log.debug("Executor is shutdown, rejecting task");
+              return;
```

Comment:
Since this is an unexpected scenario i.e. submitting a task after shutdown, it would be better to throw an exception rather than return silently.

Maybe just do `throw new RejectedExecutionException("Executor has already been shutdown");`

Reply:
I can do that, will it be a safe operation? I was worried it don't give unusual errors on a process like ingestion task, that is supposed to succeed

Reply:
If the executor has been shut down, no more tasks should be submitted to it.
If we were throwing an exception, the case that you are trying to solve (executor stuck in WAITING) would have been caught sooner.

Please do some cluster testing after making the changes.